### PR TITLE
[ActionSheet] Add properties to allow for theming

### LINKED ISSUE #4903: [ActionSheet] Expose properties to enable theming
Expose all properties to enable color, typography and shape theming.

### PR DESCRIPTION
**Pull request by codeman7**
This PR adds properties to the components to allow for theming.

Closes #4903

### CODE REVIEW

File: components/ActionSheet/src/MDCActionSheetController.m [hunk 1]
```diff
@@ -261,6 +262,13 @@ - (UITableViewCell *)tableView:(UITableView *)tableView
   cell.mdc_adjustsFontForContentSizeCategory = self.mdc_adjustsFontForContentSizeCategory;
   cell.backgroundColor = self.backgroundColor;
   cell.actionFont = self.actionFont;
+  cell.actionColor = self.actionColor;
+  for (UIView *view in cell.contentView.subviews) {
+    if ([view isKindOfClass:[UIImageView class]]) {
```
Comment (codeman7) [inline components/ActionSheet/src/MDCActionSheetController.m]: This isn't optimal but in order to set the tint color you have to wait until the view has loaded. If you do this in `layoutSubviews` of `MDCActionSheetItemTableViewCell` then it doesn't get set by default. The only other solution I could think of was a flag `imageColorHasBeenSet` in `MDCActionSheetItemTableViewCell` if that was `NO` then set the color to black.

Comment (jverkoey) [reply components/ActionSheet/src/MDCActionSheetController.m]: Can we set the tintColor on the cell's contentView? `tintColor` is a recursive API so any subviews should adopt it automatically.

File: components/ActionSheet/src/MDCActionSheetController.h [hunk 2]
```diff
@@ -138,8 +138,31 @@ __attribute__((objc_subclassing_restricted))
...
 
+/**
+ The color applied to the title of Alert Controller.
+ */
+@property(nonatomic, strong, nullable) UIColor *titleColor;
+
+/**
+ The color applied to the message of Alert Controller.
+ */
+@property(nonatomic, strong, nullable) UIColor *messageColor;
```
Comment (jverkoey) [inline components/ActionSheet/src/MDCActionSheetController.h]: As discussed in https://github.com/material-components/material-components-ios/pull/4968#discussion_r214355546, I'm a bit worried that this API may not let us build a themer that guarantees a correct theme in cases where properties are set after theming. Is there another way we can structure this API to better support the themer?

File: components/ActionSheet/src/private/MDCActionSheetItemTableViewCell.m [hunk 3]
```diff
@@ -105,6 +105,7 @@ - (void)commonMDCActionSheetItemViewInit {
   [self.contentView addSubview:_imageView];
   _imageView.translatesAutoresizingMaskIntoConstraints = NO;
   _imageView.alpha = ImageAlpha;
+  [_imageView setTintColor:[UIColor blackColor]];
```
Comment (jverkoey) [inline components/ActionSheet/src/private/MDCActionSheetItemTableViewCell.m:108]: If you remove this and rely on recursive tintColor being applied will it still work?

Comment (codeman7) [reply components/ActionSheet/src/private/MDCActionSheetItemTableViewCell.m:108]: Done

File: components/ActionSheet/src/MDCActionSheetController.h (hunk 2 above)
Comment (codeman7) [reply components/ActionSheet/src/MDCActionSheetController.h]: Think I found a solution.

File: components/ActionSheet/src/MDCActionSheetController.m (hunk 1 above)
Comment (codeman7) [reply components/ActionSheet/src/MDCActionSheetController.m]: That worked.

**Pull request closed without merging**